File: Problem Solving/LeetCode/Array 101/Max Consecutive Ones/findMaxConsecutiveOnes.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        maxCount = 0
        counter = 0
        
        for i in nums:
            if i == 1:
                counter += 1
            else:
                counter = 0
            
            if counter > maxCount:
                maxCount = counter
        
        return maxCount
        

File: Problem Solving/LeetCode/Explore/Array 101/Max Consecutive Ones II/findMaxConsecutiveOnes.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function (nums) {
    if (nums.length === 0) {
        return 0;
    }

    let maxCount = 0;
    let prevCount = 0;
    let nextCount = 0;
    for (let i = 0; i < nums.length; ++i) {
        if (nums[i] === 1) {
            ++nextCount;
        } else {
            zeroExits = true;

            maxCount = Math.max(maxCount, prevCount + nextCount);

            prevCount = nextCount + 1;
            nextCount = 0;
        }
    }

    maxCount = Math.max(maxCount, prevCount + nextCount);

    return maxCount;
};

console.log(findMaxConsecutiveOnes([1]));